packages/vite/src/node/plugins/css.ts

```diff
@@ -317,13 +320,18 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
     },
 
     async transform(raw, id) {
+      const { environment } = this
       if (
+        !environment ||
```

Comment:
I was checking unocss CI failure and it looks like unocss grabs internal `vite:css` plugin during `configResolved`, then its `transform` is reused during unocss's `renderChunk`, but `this` context during `renderChunk` is missing `environment`, so their trick is broken.

I didn't ping unocss side yet. For now, I made a patch here https://github.com/hi-ogawa/unocss/pull/1 to fix this particular issue.

Do you plan to expose `environment` to other hooks like `renderChunk`.

Reply:
For now, we were only injecting the environment in some basic hooks during build. It should probably be there for every rollup hook. We need to talk with Lukas and the Rolldown team to see if they can let us configure this in an easier way so we don't need to do costly injection hacking on our side.